bluemira/optimisation/_optimise.py

```diff
@@ -349,3 +353,26 @@ def _eq_constraint_condition(c_value: np.ndarray, tols: np.ndarray) -> np.ndarra
 def _ineq_constraint_condition(c_value: np.ndarray, tols: np.ndarray) -> np.ndarray:
     """Condition under which an inequality constraint is violated."""
     return c_value > tols
+
+
+def _set_default_termination_conditions(
+    algorithm: Union[str, Algorithm],
+    opt_conditions: Optional[Mapping[str, Optional[Union[int, float]]]] = None,
+) -> Dict[str, Union[int, float]]:
+    tols = AlgorithmDefaultTolerances()
+
+    if opt_conditions is None:
+        opt_conditions = {"max_eval": 2000}
+
+    if isinstance(algorithm, str):
+        algorithm = Algorithm[algorithm]
+
+    if not isinstance(algorithm, Algorithm):
+        return opt_conditions
+
+    # If the value of any condition is None it is dropped
+    return {
+        k: v
+        for k, v in {**getattr(tols, algorithm.name).to_dict(), **opt_conditions}.items()
+        if v is not None
+    }
```

Comment:
I'd envisioned that if `opt_conditions` were given by the user, we completely discard the defaults. If I'd specified `ftol_rel = 1e-8` I might be a little peeved if `COBYLA` stopped at `ftol_abs = 1e-4`